test/test_console_tool.py

```diff
@@ -566,6 +566,183 @@ def test_get_bucket_show_size(self):
+        bucket.upload(UploadSourceBytes(b'test'), 'test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/4/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/4/5/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/4/5/6/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/4/5/6/7/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/4/5/6/7/8/test')
+        bucket.upload(UploadSourceBytes(b'test'), '1/2/3/4/5/6/7/8/9/test')
+        bucket.upload(UploadSourceBytes(b'check'), 'check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/4/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/4/5/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/4/5/6/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/4/5/6/7/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/4/5/6/7/8/check')
+        bucket.upload(UploadSourceBytes(b'check'), '1/2/3/4/5/6/7/8/9/check')
+
+        # Now check the output of get-bucket against the canon.
+        expected_stdout = '''
+        {
+            "accountId": "my-account",
+            "bucketId": "bucket_0",
+            "bucketInfo": {},
+            "bucketName": "my-bucket",
+            "bucketType": "allPublic",
+            "corsRules": [],
+            "fileCount": 20,
+            "lifecycleRules": [],
+            "revision": 1,
+            "totalSize": 90
+        }
+        '''
+        self._run_command(['get-bucket', '--showSize', 'my-bucket'], expected_stdout, '', 0)
+
+    def test_get_bucket_with_hidden(self):
+        self._authorize_account()
+        self._create_my_bucket()
+
+        # Put some files into the test bucket. Unroll the loop for convenience.
+        bucket = self.b2_api.get_bucket_by_name('my-bucket')
+        bucket.upload(UploadSourceBytes(b'test'), 'upload1')
+        bucket.upload(UploadSourceBytes(b'test'), 'upload2')
+        bucket.upload(UploadSourceBytes(b'test'), 'upload3')
+        bucket.upload(UploadSourceBytes(b'test'), 'upload4')
+        bucket.upload(UploadSourceBytes(b'test'), 'upload5')
+        bucket.upload(UploadSourceBytes(b'test'), 'upload6')
+
+        # Hide some new files. Don't check the results here; it will be clear enough that
+        # something has failed if the output of 'get-bucket' does not match the canon.
+        stdout, stderr = self._get_stdouterr()
+        console_tool = ConsoleTool(self.b2_api, stdout, stderr)
+        console_tool.run_command(['b2', 'hide_file', 'my-bucket', 'hidden1'])
```

Comment:
When people hide files, the normal use case is to hide a file that already exists.  Uploading a hide marker for a file that does exist would be pointless.

In fact, it looks like this is a case where the simulator is inaccurate.  The real B2 service will refuse to hide a file that isn't there:

```
$ b2 hide-file bwb-test2 e
ERROR: File not present: e
```

Reply:
The simulator thinks that hiding files creates new files. When I originally added `n` files and then hid `n` files, it thought that `2n` files were in existence. I don't have enough experience to identify differences between the simulator and the service a priori. I assumed that the simulator was "good enough". It sounds like this test exposes a problem with the simulator that should be addressed to improve verisimilitude.

Reply:
Yep.  Not your fault the simulator is wrong.  I'm the one who wrote it.  :-/

Hiding a file adds a version of the file that is a "hide marker".  The original version is still there, and there is a zero-size hide marker.  The [docs](https://www.backblaze.com/b2/docs/file_versions.html) talk about this some.